### DISCUSSION TITLE
Users stories

### DISCUSSION THREAD

**Discussion by bartoszptak** [PUTvision/qgis-plugin-deepness#86] [Show and tell]
**Let us know, how you use the Deepness plugin in your application/research!**

For example:
- How and where you are using Deepness models from the registry?
- Did you train your own model for the Deepness plugin? In which task, dataset, or architecture?

**Comment by blxl909** [comment]
I train my own model in semantic segmentation, LoveDA dataset. My model has some operators that onnx didn't support, so I implement them myself. besides, my data preprocessing and postprocessing steps are different from the plugin, so I modify them after that, model runs well. I really like this plugin and the plugin source code, it's easy to use and modify ( that means developer-friendly  (*^▽^*) )

**Comment by fredrikbacke92** [comment]
I exported tiles with deepness, annotated and trained a object detection model using roboflow and ultralytics. Exported it to onnx-file but there was an error. After some tweaks the maintainer fixed the problem and got it to work!

**Reply by Leonn714** [reply]
Would love to know how you did this.  I am following the same workflow and was only able to get the detector model to work but would not work with segmentation model

**Comment by tenentedan86** [comment]
Dear All,
Approximately five to six months ago, I began utilizing the Deepness plugin to develop an efficient method for the rapid identification of sea cucumbers in Mediterranean waters. These organisms are significant components of soft-bottom ecosystems, as they play a crucial role in influencing community ecology by offering essential ecosystem services, including sediment bioturbation and organic matter recycling. My research involves seabed mapping through underwater photogrammetry, which has allowed me to create extensive maps with high spatial resolution, facilitating the detection of sea cucumbers.
Initially, I had little knowledge of the underlying principles of YOLO; however, I successfully trained my first model by following the highly informative tutorial by Fredrick titled "Deploy Object Detection Model with Deepness Without a Single Line of Code." Collaborating with a colleague (Gian Mario) who possesses greater expertise in Python, we opted to construct and train a more sophisticated model without relying on Ultralytics, thereby ensuring a high level of customization, which included the implementation of augmentation techniques.
The final model was subsequently implemented in Deepness across five extensive mapping campaigns, yielding intriguing data and significantly favorable outcomes in comparison to manual counts. A comprehensive description of this work will be provided in a forthcoming paper that is currently in progress. A couple of days ago, a recent update (refer to https://github.com/PUTvision/qgis-plugin-deepness/issues/66) enabled the estimation of confidence levels for each detection, which could significantly enhance the filtering of results. I believe that this tool, along with the overall innovative concept developed by the team, has equipped QGIS users with a highly effective resource for a wide range of research applications. 
We hope that sharing our experiences will contribute to future developments, and we are always open to further discussions and ideas.
 
All the best,
Daniele and Gian Mario
